src/main/java/org/apache/sling/caconfig/impl/ConfigurationBindingsValueProvider.java

```diff
@@ -47,7 +51,14 @@
  */
 @Component(immediate = true, service = BindingsValuesProvider.class, property = {
         "javax.script.name=sightly",
-        Constants.SERVICE_RANKING + "=100" })
+        Constants.SERVICE_RANKING + "=100"
+},
+reference = {
```

Comment:
i suggest creating a new management "multiplexer" interface for the use case of getting the result resource of multiple implementations of the SPI interface - see package "org.apache.sling.caconfig.management.multiplexer" and the implementation. thus the logic can be re-used e.g. for [SLING-7256|https://issues.apache.org/jira/browse/SLING-7256], and a bit of logic is away from this class.

Reply:
Just added the multiplexer for ConfigurationBindingsResourceDetectionStrategy